### PR TITLE
Add layout specific injections

### LINKED ISSUE #101: Add specify a layout in injections
### What steps will reproduce the problem?
I have 2 layouts main and second. For main in the view, I need attributes to display, for example, the number of online users, the number of new users, etc. For the second layout I don’t need these parameters, but some others may be needed

### What is the expected result?
I would like it to be possible in injections to specify the layout for which this injection needs to be connected.

### Example

```php
readonly final class CommonViewInjection implements CommonParametersInjectionInterface
{
    public function __construct(
        private ApplicationParameters $applicationParameters,
        private UrlGeneratorInterface $urlGenerator,
        private CurrentUser $user
    ) {
    }

    public function getCommonParameters(): array
    {
        return [
            'applicationParameters' => $this->applicationParameters,
            'urlGenerator' => $this->urlGenerator,
            'user' => $this->user,
        ];
    }

    public function includeLayout(): array
    {
      // return ['main', 'second']; // inject only for 'main', 'second' layout
        return ['*']; // wild for all layout OR empty
    }

    public function excludeLayout(): array
    {
        //case when i have more 30 layout and wont exclude this inject for only one
        return ['other']; // exclude layout for inject
    }
}
```

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | Fix #101

### CODE REVIEW

File: src/ViewRenderer.php [hunk 1]
```diff
@@ -348,10 +345,8 @@ private function renderProxy(
     private function getCommonParameters(): array
     {
         $parameters = [];
-        foreach ($this->injections as $injection) {
-            if ($injection instanceof CommonParametersInjectionInterface) {
-                $parameters = array_merge($parameters, $injection->getCommonParameters());
-            }
+        foreach ($this->getInjections($this->layout, CommonParametersInjectionInterface::class) as $injection) {
+            $parameters = array_merge($parameters, $injection->getCommonParameters());
```
Comment (samdark) [inline src/ViewRenderer.php:349]: `array_merge` within `foreach()` isn't great.

See https://www.exakat.io/en/speeding-up-array_merge/

File: src/ViewRenderer.php [hunk 2]
```diff
@@ -366,10 +361,8 @@ private function getCommonParameters(): array
     private function getLayoutParameters(): array
     {
         $parameters = [];
-        foreach ($this->injections as $injection) {
-            if ($injection instanceof LayoutParametersInjectionInterface) {
-                $parameters = array_merge($parameters, $injection->getLayoutParameters());
-            }
+        foreach ($this->getInjections($this->layout, LayoutParametersInjectionInterface::class) as $injection) {
+            $parameters = array_merge($parameters, $injection->getLayoutParameters());
```
Comment (samdark) [inline src/ViewRenderer.php:365]: Same here.

File: src/ViewRenderer.php [hunk 3]
```diff
@@ -382,10 +375,8 @@ private function getLayoutParameters(): array
     private function getMetaTags(): array
     {
         $tags = [];
-        foreach ($this->injections as $injection) {
-            if ($injection instanceof MetaTagsInjectionInterface) {
-                $tags = array_merge($tags, $injection->getMetaTags());
-            }
+        foreach ($this->getInjections($this->layout, MetaTagsInjectionInterface::class) as $injection) {
+            $tags = array_merge($tags, $injection->getMetaTags());
```
Comment (samdark) [inline src/ViewRenderer.php:379]: Same here.

File: src/ViewRenderer.php [hunk 4]
```diff
@@ -398,12 +389,34 @@ private function getMetaTags(): array
     private function getLinkTags(): array
     {
         $tags = [];
+        foreach ($this->getInjections($this->layout, LinkTagsInjectionInterface::class) as $injection) {
+            $tags = array_merge($tags, $injection->getLinkTags());
```
Comment (samdark) [inline src/ViewRenderer.php:393]: Same here.

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/ViewRenderer.php (hunk 1 above)
Comment (vjik) [reply src/ViewRenderer.php]: Done

**Review by samdark — APPROVED** [review]

**Pull request merged**